src/components/board/grid.ts

```diff
@@ -21,3 +27,37 @@ export const useGridCenter: (loc: GridLoc) => Point = ({ gridX, gridY }) => {
   const gridSize = useGridSize();
   return { x: (gridX + 0.5) * gridSize, y: (gridY + 0.5) * gridSize };
 };
+
+export const windowToGridLoc: (
+  windowPoint: Point,
+  gridSize: number,
+  boardTopLeft: Point
+) => GridLoc = ({ x, y }, gridSize, { x: boardX, y: boardY }) => {
+  return {
+    gridX: Math.floor((x + boardX) / gridSize),
+    gridY: Math.floor((y + boardY) / gridSize),
+  };
+};
+
+export const droppedOnGrid: (
```

Comment:
This function was tricky. As far as I can tell, the target location that Konva reports on a drag event is relative to the shape's original position. So I had to shift it both by where it was originally and by the window pan position to arrive at the "physical" coordinates and then translate that into a `GridLoc`.